tests/integration/dynamic/nodejs-config/index.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright 2016-2023, Pulumi Corporation.
+
+import * as pulumi from "@pulumi/pulumi";
+
+class CustomResource extends pulumi.dynamic.Resource {
+    public readonly authenticated!: pulumi.Output<string>;
+    public readonly color!: pulumi.Output<string>;
+
+    constructor(name: string, opts?: pulumi.ResourceOptions) {
+        super(
+            new DummyResourceProvider(),
+            name,
+            {
+                authenticated: undefined,
+                color: undefined
+            },
+            opts,
+            "custom-provider",
+            "CustomResource",
+        );
+    }
+}
+
+class DummyResourceProvider implements pulumi.dynamic.ResourceProvider {
+    private password: string;
+    private color: string;
+
+    async configure(req: pulumi.dynamic.ConfigureRequest): Promise<any> {
+        this.password = req.config.require("password");
+        this.color = req.config.get("colors:banana") || "blue";
```

Comment:
Not sure whether we have thoughts on this/whether it's possible depending on Node versions we support but I wonder if we should think about encouraging `??` instead of `||` for defaulting in the presence of `undefined` (assuming that this is the behaviour we want here).

Reply:
Ah `??` is already in TypeScript 3.7, so I think we can use it everywhere in our code. This test uses a TS 5 in any case, so we're good.